java/clients/src/main/java/sleeper/clients/deploy/documentation/GenerateJarsDocumentation.java

```diff
@@ -0,0 +1,115 @@
+import sleeper.core.deploy.DockerDeployment;
+import sleeper.core.deploy.LambdaJar;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.PrintStream;
+import java.io.UncheckedIOException;
+import java.io.UnsupportedEncodingException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.util.List;
+import java.util.function.Consumer;
+
+public class GenerateJarsDocumentation {
+
+    private GenerateJarsDocumentation() {
+    }
+
+    public static void main(String[] args) throws IOException {
+        Path path = Path.of(args[0]).resolve("docs/deployment/jars-to-upload.md");
+        if (Files.exists(path)) {
+            Files.delete(path);
+        }
+        Files.createFile(path);
+        writeFile(path, output -> writePropertiesMarkdownFile(output, "Deployment Jars", "These are the docker deployment Jars",
+                createDockerDeploymentTableWriter(DockerDeployment.all())));
+        writeFile(path, output -> writePropertiesMarkdownFile(output, "Lambda Jars", "These are the Lambda deploy jars",
+                createLambdaJarTableWriter(LambdaJar.ALL)));
+    }
+
+    private static TableWriter createDockerDeploymentTableWriter(List<DockerDeployment> deployments) {
+        TableFieldDefinition name = TableFieldDefinition.field("Property Name");
+        TableFieldDefinition optionalStack = TableFieldDefinition.field("Optional Stack");
+        TableFieldDefinition multiplatform = TableFieldDefinition.field("Multiplatform");
+
+        TableWriterFactory factory = TableWriterFactory.builder()
+                .structure(TableStructure.MARKDOWN_FORMAT)
+                .addFields(name, optionalStack, multiplatform)
+                .build();
+
+        return factory.tableBuilder()
+                .itemsAndWriter(deployments, (deployment, row) -> {
+                    row.value(name, deployment.getDeploymentName());
+                    row.value(optionalStack, deployment.getOptionalStack());
+                    row.value(multiplatform, deployment.isMultiplatform());
+                }).build();
+    }
+
+    private static TableWriter createLambdaJarTableWriter(List<LambdaJar> deployments) {
+        TableFieldDefinition fileName = TableFieldDefinition.field("File Name");
+        TableFieldDefinition imageName = TableFieldDefinition.field("Image Name");
+        TableFieldDefinition isAlwaysDockerDeploy = TableFieldDefinition.field("Always docker deploy");
+
+        TableWriterFactory factory = TableWriterFactory.builder()
+                .structure(TableStructure.MARKDOWN_FORMAT)
+                .addFields(fileName, imageName, isAlwaysDockerDeploy)
+                .build();
+
+        return factory.tableBuilder()
+                .itemsAndWriter(deployments, (deployment, row) -> {
+                    row.value(fileName, deployment.getFilename());
```

Comment:
This filename includes the version number. We can expect people will read the documentation on the develop branch, because that's the default branch. It seems a little confusing to see a snapshot version there.

Might it be worth refactoring LambdaJar to retain the format string and let you get the filename with a different version number, so we can put some placeholder for the version in the documentation instead?